includes/polling/ports.inc.php

```diff
@@ -274,8 +288,41 @@
 }//end if
 
 if ($config['enable_ports_poe']) {
-    $port_stats = snmpwalk_cache_oid($device, 'pethPsePortEntry', $port_stats, 'POWER-ETHERNET-MIB');
-    $port_stats = snmpwalk_cache_oid($device, 'cpeExtPsePortEntry', $port_stats, 'CISCO-POWER-ETHERNET-EXT-MIB');
+    // Code by OS device
+    if ($device['os'] == 'procurve') {
+        // Not tested
+        $port_stats_poe = snmpwalk_cache_oid($device, 'pethPsePortEntry', array(), 'POWER-ETHERNET-MIB');
+    }
+
+    if ($device['os'] == 'ios') {
+        $port_stats_poe = snmpwalk_cache_oid($device, 'pethPsePortEntry', array(), 'POWER-ETHERNET-MIB');
+        $port_stats_poe = snmpwalk_cache_oid($device, 'cpeExtPsePortEntry', $port_stats_poe, 'CISCO-POWER-ETHERNET-EXT-MIB');
+        $port_ent_to_if = snmpwalk_cache_oid($device, 'portIfIndex', array(), 'CISCO-STACK-MIB');
```

Comment:
This + the extra walks below is going to slow polling down and could do so considerably. What's the reason for performing these?

Reply:
'pethPsePortEntry' is now removed as it is not necessary for Cisco polling. cpeExtPsePortEntry and portIfIndex are mandatory : 
cpeExtPsePortEntry contains the actual value, but indexed by entIndex, and portIfIndex contains the mapping to change entIndex into ifIndex.
Then we have a foreach loop to do the actual mapping and get the usable values.